homeassistant/components/climate/generic_thermostat.py

```diff
@@ -278,33 +278,23 @@ def max_temp(self):
         # Get default temp from super class
         return super().max_temp
 
-    @asyncio.coroutine
-    def _async_sensor_changed(self, entity_id, old_state, new_state):
+    async def _async_sensor_changed(self, entity_id, old_state, new_state):
         """Handle temperature changes."""
         if new_state is None:
             return
 
-        self._async_update_temp(new_state)
-        self._async_control_heating()
-        yield from self.async_update_ha_state()
+        await self._async_update_temp(new_state)
+        await self._async_control_air_conditioner()
+        await self.async_update_ha_state()
 
-    @callback
-    def _async_switch_changed(self, entity_id, old_state, new_state):
-        """Handle heater switch state changes."""
+    async def _async_switch_changed(self, entity_id, old_state, new_state):
+        """Handle A/C switch state changes."""
         if new_state is None:
             return
-        self.async_schedule_update_ha_state()
 
-    @callback
-    def _async_keep_alive(self, time):
-        """Call at constant intervals for keep-alive purposes."""
-        if self._is_device_active:
-            self._heater_turn_on()
-        else:
-            self._heater_turn_off()
+        await self.async_update_ha_state()
 
-    @callback
-    def _async_update_temp(self, state):
+    async def _async_update_temp(self, state):
```

Comment:
Why is this now async?

Reply:
It works as either `async` or `callback` as it used to be. I marked the methods that can be asynchronous as `async` in order to:
* make the code more consistent (avoid mixing async and regular code) and enforce its asynchronicity in the future
* make it consistent with the styling of having all the functions/methods named `async_???` be actually asynchronous - this way, people working on the code will see a method name that starts with `async`, and will be able to safely assume that they can `await` on that method.

Reply:
I understand the desire to not mix styles. Checking this again, though, it appears that you forgot to add `await` to a call on line 147. Doing so will not be easy if you want to maintain the current consistency. So you might want to reconsider :)

Note that the usual Home Assistant convention is that functions prefixed with `async_` must be run in the event loop (i.e. not in worker threads). It does _not_ mean they are necessarily `async def`. So it is fine to name callbacks `async_`, even if it is confusing.